spec/models/service_order_spec.rb

```diff
@@ -1,68 +1,81 @@
   let(:admin)         { FactoryGirl.create(:user_with_group, :userid => "admin") }
-  let(:service_order) { FactoryGirl.create(:service_order, :state => 'cart', :user => user, :tenant => tenant) }
+  let(:service_order) do
+    FactoryGirl.create(:service_order, :state  => ServiceOrder::STATE_CART,
+                                       :user   => user,
+                                       :tenant => tenant)
+  end
   let(:user)          { FactoryGirl.create(:user_with_group, :tenant => tenant) }
   let(:ems)           { FactoryGirl.create(:ems_vmware) }
   let(:vm)            { FactoryGirl.create(:vm_vmware, :name => "vm1", :location => "abc/def.vmx") }
   let(:vm_template)   { FactoryGirl.create(:template_vmware, :name => "template1", :ext_management_system => ems) }
-  let(:request) do
-    FactoryGirl.create(:service_template_provision_request,
-                       :description => 'Service Request',
-                       :requester   => admin,
-                       :options     => {:service_order_id => service_order.id})
-  end
-  let(:request2) do
-    FactoryGirl.create(:service_template_provision_request,
-                       :description => 'Service Request',
-                       :requester   => admin,
-                       :options     => {:service_order_id => service_order.id})
-  end
-  let(:request3) do
-    FactoryGirl.create(:service_template_provision_request,
-                       :description => 'Service Request',
-                       :requester   => admin,
-                       :options     => {:service_order_id => service_order.id})
-  end
+  let(:request)       { create_request }
+  let(:request2)      { create_request }
+  let(:request3)      { create_request }
 
   it "should add an miq_request properly" do
-    request
-
     expect request.service_order == service_order.id
     expect request.process == false
-    expect(ServiceOrder.first).to have_attributes(
-      :name  => 'service order',
-      :state => 'cart'
-    )
+    expect(service_order.state).to eq(ServiceOrder::STATE_CART)
   end
 
   it "should add multiple miq_requests properly" do
     [request, request2, request3].each do |r|
       expect r.service_order == service_order.id
       expect r.process == false
     end
-    service_order = ServiceOrder.first
-    expect(service_order).to have_attributes(
-      :name  => 'service order',
-      :state => 'cart'
-    )
+    expect(service_order.state).to eq(ServiceOrder::STATE_CART)
     expect service_order.miq_requests.count == MiqRequest.count
   end
 
-  it "should checkout properly" do
-    r1 = request
-    r2 = request2
-    r3 = request3
+  it "should checkout properly for cart service order" do
+    service_order.update_attributes(:miq_requests => [request, request2, request3])
     service_order.checkout
-    [r1, r2, r3].each do |r|
-      r.reload
+    [request, request2, request3].each do |r|
       expect r.service_order == service_order.id
```

Comment:
I think it's annoying that this works (service_order == id, it should == object) :)  It would be nicer to see this outside the block like this:
```ruby
expect(service_order.miq_requests).to match_array([request, request2, request3])
```
But i'm also tempted to remove the expectation because it's really just testing the Rails `has_many` relation.